grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1132,106 +1140,438 @@ class StockMovementService {
-                            "The stock entry you selected: lot \"${lotNumberName}\" and bin location \"${binLocationName}\" is not available. Please review pick."
+                            "importPickCommand.requisitionItem.notFound.error",
+                            [data.id] as Object[],
+                            "Requisition item id: ${data.id} not found"
                     )
                 }
+                if (pickPageItem) {
+
+                    // TODO Add guidance for developers - we ALWAYS need to resolve to an inventory
+                    //  item when we're working with a lot number.
+                    // Resolve given lot number to an inventory item
+                    InventoryItem inventoryItem =
+                            pickPageItem.requisitionItem.product.getInventoryItem(data.lotNumber, data.expirationDate)
 
-                Integer itemQuantitySum = picklistItemsGroupedByRequisitionItem[data.id].sum { it.quantity }
-                if (itemQuantitySum > pickPageItem.requisitionItem.quantity) {
-                    Boolean allowsOverPick = stockMovement.origin.supports(ActivityCode.ALLOW_OVERPICK)
-                    if (!allowsOverPick) {
+                    // If there's no inventory item for the provide lot number, we throw a validation error
+                    if (!inventoryItem) {
+                        // FIXME Add error code to messages.properties
+                        data.errors.rejectValue(
+                                "lotNumber",
+                                "importPickCommand.inventoryItem.notFound.error",
+                                [data.lotNumber] as Object[],
+                                "Unable to find inventory item with lot number {0}")
+                    }
+                    else {
+
+                        // If there is an inventory item found, let's check to see if there's more
+                        // than one picklist item with an empty bin
+                        Integer countEmptyBinLocations = picklistItems.findAll { it.lotNumber == inventoryItem.lotNumber }.count { !it.binLocation }
+                        if (countEmptyBinLocations > 1) {
+                            data.errors.rejectValue(
+                                    "binLocation",
+                                    "importPickCommand.binLocation.multipleEmpty.error",
+                                    [data.lotNumber] as Object[],
+                                    "Bin location cannot be empty for multiple rows with the same lot number {0}"
+                            )
+                        }
+                    }
+
+                    // OBPIH-6331 Resolve bin location, if a bin location value was provided. Otherwise,
+                    // if the user-provided bin location is null or ambiguous then we need to apply
+                    // allocation rules
+                    Location internalLocation = command.location.getInternalLocation(data.binLocation)
```

Comment:
Is it technically possible (now or in the future, through API) that someone will be able to import an outbound not from the "current location"? This is why I am wondering if we should not start using origin instead of current location as a `command.location` :thinking: